pkg/beacon/relay/groupselection/groupselection.go

```diff
@@ -24,25 +33,28 @@ type Result struct {
 	GroupSelectionEndBlock uint64
 }
 
-// CandidateToNewGroup attempts to generate and submit tickets for the staker to
-// join a new group.
-//
-// There are two phases of ticket submission:
-// - initial ticket submission,
-// - reactive ticket submission.
+// CandidateToNewGroup attempts to generate and submit tickets for the
+// staker to join a new group.
 //
-// During the initial ticket submission, only tickets with a value below the
-// natural threshold are submitted to the chain. Those tickets have the highest
-// chance of being selected to the group and this way we minimize staker's
-// gas expenditure.
+// To minimize the submitter's cost by minimizing the number of redundant
+// tickets that are not selected into the group, tickets are submitted in
+// 11 rounds, each round taking 3 blocks.
```

Comment:
Actually... we compute the number of rounds. We should say `tickets are submitted in N rounds` and later:
```
// - in round 0, tickets with N or more leading zeros are submitted
// - in round 1, tickets with N-1 or more leading zeros are submitted
// (...)
// - in round N, tickets with no leading zeros are submitted.
```